ext/drx/drx.h

```diff
@@ -435,6 +435,19 @@ DR_EXPORT
 size_t
 drx_buf_get_buffer_size(void *drcontext, drx_buf_t *buf);
 
+
+DR_EXPORT
+/**
+ * Pads a basic block with a label for routines which rely on inserting instrumentation
+ * after every instruction. Note that users of this routine must act on the previous
+ * instruction before the is-app check afterwards, because the label is not marked as an
+ * app instruction.
+ *
+ * \returns whether terminator was introduced.
```

Comment:
Yes, DR requires a syscall or cbr or mbr to end a block to avoid nasty complexities in its core operation, but I'm not sure about a label being after: it might allow it (or we could make sure it allows it...)